plugins/modules/writable_snapshot.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python
+# Copyright: (c) 2024, Dell Technologies
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+""" Ansible module for managing writable snapshots on PowerScale"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: writable_snapshot
+version_added: '3.3.0'
+short_description: Manage writable snapshots on PowerScale
+description:
+- You can perform the following operations.
+- Managing snapshots on PowerScale.
+- Create a writable snapshot.
+- Delete a writable snapshot.
+
+extends_documentation_fragment:
+  - dellemc.powerscale.powerscale
+
+author:
+- Kritika Bhateja(@Kritika-Bhateja-03) <ansible.team.dell.com>
+
+options:
+  writable_snapshot:
+    description:
+    - List of writable snapshot details.
+    required: False
+    type: list
+    elements: dict
+    suboptions:
+      state:
+        description:
+        - The state of the writable snapshot to create or delete.
+        - I(state) is C(present) - To create a writable snapshot.
```

Comment:
what about get operation?

Reply:
As per spec , I see only create and delete are supported.
Also, get operation would be a part of info module.